src/engine/sparqlExpressions/SparqlExpressionValueGetters.cpp

```diff
@@ -145,9 +145,23 @@ IntDoubleStr ToNumericValueGetter::operator()(
       return id.getInt();
     case Datatype::Double:
       return id.getDouble();
-    default:
-      return std::monostate{};
+    case Datatype::Bool:
+      return static_cast<int>(id.getBool());
+    case Datatype::Undefined:
+    case Datatype::VocabIndex:
+    case Datatype::LocalVocabIndex:
+    case Datatype::TextRecordIndex:
+    case Datatype::WordVocabIndex:
+    case Datatype::Date:
+    case Datatype::BlankNodeIndex:
+      auto optString = LiteralFromIdGetter{}(id, context);
+      if (optString.has_value()) {
+        return optString.value();
```

Comment:
```suggestion
        return std::move(optString.value());
```
Unfortunately necessary to avoid the copy.